benchmark/SortedJoinBenchmark.cpp

```diff
@@ -0,0 +1,97 @@
+// Copyright 2015, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Andre Schlegel (November of 2022, schlegea@informatik.uni-freiburg.de)
+// Author of the file this file is based on: Björn Buchhold (buchhold@informatik.uni-freiburg.de)
+
+#include <gtest/gtest.h>
+
+#include <cstdio>
+#include <algorithm>
+
+#include "engine/Engine.h"
+#include "engine/Join.h"
+#include "engine/QueryExecutionTree.h"
+#include "util/Forward.h"
+#include "engine/idTable/IdTable.h"
+#include "../benchmark/Benchmark.h"
+
+// Local helper function.
+#include "../benchmark/util/IdTableHelperFunction.h"
+
+// Non-local helper function.
+#include "../test/util/IdTableHelpers.h"
+#include "../test/util/JoinHelpers.h"
+#include "../test/IndexTestHelpers.h"
+
+// Benchmarks for sorted tables, with and without overlapping values in
+// IdTables. Done with normal join and hash join.
+void BM_SortedIdTable(BenchmarkRecords* records) {
+  // For easier changing of the IdTables size.
+  // Important: Apparently Join::join has an inbuild timeout checker, which
+  // keeps getting triggerd by tables with a size between 500-600, or higher.
+  // This leads to a segmentation fault, because it keeps trying to throw an
+  // exception about the time out, but that exception tries to include
+  // information about the query execution tree using a shared pointer. And, as
+  // far as I can tell, that pointer was never actually initialized here, so
+  // it points into memory, which we are not allowed to access. Which leads
+  // to the seg fault.
+  const size_t NUMBER_ROWS = 500;
+  const size_t NUMBER_COLUMNS = 500; 
```

Comment:
Isn't this fixed by now and you can add a range of larger benchmarks here?

Reply:
It wasn't, the last time I looked, but I will take another look.

Reply:
After merging with the main repo ([ff09aef](https://github.com/ad-freiburg/qlever/pull/860/commits/ff09aef8b2e7e3ce145e92d901e37b4f2ad8e190)), which, hopefully, didn't break anything, or undo any positive changes, the benchmarks seem to no longer crash with increased range.

I increased them in [4df1d95](https://github.com/ad-freiburg/qlever/pull/860/commits/4df1d95f0878335954bea086d3de4682448d2a68) and I'm locally checking, if they run through. (Which seems to take a while.)

Reply:
Alright, locally doesn't seem to work anymore. It seems, that my Ubuntu kills it after a while, because it uses to much ressources.
I will try to run it on a remote machine and write a new comment with my results.

Reply:
I had to lower the size of the tables ([7c3a458](https://github.com/ad-freiburg/qlever/pull/860/commits/7c3a458a344cb41d326ee3e4c85cc1fcc89f4ad1)), but now it runs.

The problems seems to be fixed now, based on this field test and the only remaining constraint seems to be computer power.